Add an overview for monitoring tutorials (#5691)
kyma-project/kyma: docs/monitoring/08-01-overview.md

---
title: Overview
type: Tutorials
---

> The set of monitoring tutorials you are about to read describes the complete monitoring flow for your metrics in Kyma. Going through the tutorials, you will get the gist of Kyma in-built monitoring applications, such as Prometheus, Grafana, and Alertmanager. You will also understand how and where you can observe and visualize your service metrics to monitor them for any alerting values.

Reviewer: I would maybe say `monitoring flow for your service` rather than metrics. 🤔 as we use the service to expose metrics that are later on picked by Prometheus.

Author: sure --> `monitoring flow for your services in Kyma.`?

> The set of monitoring tutorials you are about to read describes the complete monitoring flow for your services in Kyma. Going through the tutorials, you will get the gist of Kyma in-built monitoring applications, such as Prometheus, Grafana, and Alertmanager. You will also understand how and where you can observe and visualize your service metrics to monitor them for any alerting values.

Author: Sure 👍

> All tutorials use the [`monitoring-custom-metrics`](https://github.com/kyma-project/examples/tree/master/monitoring-custom-metrics) example and one of its services called `sample-metrics-8081`. This service exposes the `cpu_temperature_celsius` custom metric on the `/metrics` endpoint. This custom metric is the central element of all tutorials. Its value simulates the current processor temperature and changes randomly from 60 to 90 degrees Celsius. The alerting threshold in these tutorials is 75 degrees Celsius. If the temperature exceeds this value, the Grafana dashboard, Prometheus rule, and Alertmanager notifications you create clearly inform you about this.

Author: I added "the" at the beginning but rephrased a bit the second part not to repeat it one more time: `This custom metric is the central element of the whole tutorial set.` Hope that's ok as well?

> All tutorials use the [`monitoring-custom-metrics`](https://github.com/kyma-project/examples/tree/master/monitoring-custom-metrics) example and one of its services called `sample-metrics-8081`. This service exposes the `cpu_temperature_celsius` custom metric on the `/metrics` endpoint. This custom metric is the central element of all tutorials. Its value simulates the current processor temperature and changes randomly from 60 to 90 degrees Celsius. The alerting threshold in these tutorials is 75 degrees Celsius. If the temperature exceeds this value, the Grafana dashboard, Prometheus rule, and Alertmanager notifications you create clearly inform you of the alerting value.

Reviewer: ```suggestion
All tutorials use the [`monitoring-custom-metrics`](https://github.com/kyma-project/examples/tree/master/monitoring-custom-metrics) example and one of its services called `sample-metrics-8081`. This service exposes the `cpu_temperature_celsius` custom metric on the `/metrics` endpoint. This custom metric is the central element of all tutorials. Its value simulates the current processor temperature and changes randomly from 60 to 90 degrees Celsius. The alerting threshold in these tutorials is 75 degrees Celsius. If the temperature exceeds this value, the Grafana dashboard, Prometheus rule, and Alertmanager notifications you create clearly inform you about this.
```
I wouldn't say that they inform you about the value itself (although it's mentioned), but about the fact that the temp value was exceeded.

The tutorial set consists of these documents:

1. [**Observe application metrics**](#tutorials-observe-application-metrics) in which you redirect the `cpu_temperature_celsius` metric to the localhost and the Prometheus UI. You later observe how the metric value changes in the predefined 10 seconds interval in which Prometheus scrapes the metric values from the service's `/metrics` endpoint.

> 2. [**Create a Grafana dashboard**](#tutorials-create-a-grafana-dashboard) in which you create a Grafana dashboard of a Gauge type for the `cpu_temperature_celsius` metric. This dashboard shows explicitly when the CPU temperature exceeds the predefined threshold of 75 degrees Celsius and turns red whenever this value is higher.

Author: Good point 👍

> 2. [**Create a Grafana dashboard**](#tutorials-create-a-grafana-dashboard) in which you create a Grafana dashboard of a Gauge type for the `cpu_temperature_celsius` metric. This dashboard shows explicitly when the CPU temperature exceeds the predefined threshold of 75 degrees Celsius and turns red whenever this value is higher.

Author: I changed it into `This dashboard shows explicitly when the CPU temperature is equal to or exceeds the predefined threshold of 75 degrees Celsius, at which point the dashboard turns red.` as the ending of the sentences `turns red whenever this value is higher.` was also misleading.

> 3. [**Define alerting rules**](#tutorials-define-alerting-rules) in which you define the `CPUTempHigh` alerting rule by creating a PrometheusRule resource. Prometheus accesses the `/metrics` endpoint every 10 seconds and validates the current value of the `cpu_temperature_celsius` metric. If the value is higher than 75 degrees Celsius, Prometheus waits for 10 seconds and triggers the rule. You can observe the rule on the Prometheus dashboard.

Reviewer: ```suggestion
3. [**Define alerting rules**](#tutorials-define-alerting-rules) in which you define the `CPUTempHigh` alerting rule by creating a PrometheusRule resource. Prometheus accesses the `/metrics` endpoint every 10 seconds and validates the current value of the `cpu_temperature_celsius` metric. If the value is higher than 75 degrees Celsius, Prometheus waits for 10 seconds to check it again. If the value still exceeds the threshold, it fires the alarm. You can observe the rule and the alert on the Prometheus dashboard. 
```
I changed the last two sentences a bit :)

> 3. [**Define alerting rules**](#tutorials-define-alerting-rules) in which you define the `CPUTempHigh` alerting rule by creating a PrometheusRule resource. Prometheus accesses the `/metrics` endpoint every 10 seconds and validates the current value of the `cpu_temperature_celsius` metric. If the value is higher than 75 degrees Celsius, Prometheus waits for 10 seconds to check it again. If the value still exceeds the threshold, Prometheus triggers the rule. You can observe both the rule and the alert it generates on the Prometheus dashboard.

Author: unfortunately, in that case it is defined only for values higher than 75

> 4. [**Send notifications to Slack**](#tutorials-send-notifications-to-slack) in which you configure Alertmanager to send notifications on Prometheus alerts to a Slack channel. This way, whenever Prometheus triggers or resolves the `CPUTempHigh` alert, Alertmanager sends a notification to the `test-monitoring-alerts` Slack channel.

Reviewer: ```suggestion
4. [**Send notifications to Slack**](#tutorials-send-notifications-to-slack) in which you configure Alertmanager to send notifications on Prometheus alerts to a Slack channel. This way, whenever Prometheus triggers or resolves the `CPUTempHigh` alert, Alertmanager sends a notification to the `test-monitoring-alerts` Slack channel defined for this purpose.
```
We can maybe add the info that our Slack channel is a sample one (I added `defined for this purpose`, but it may be something different).

> See the diagram for an overview of tutorials purpose and tools used in them:

Reviewer: Is the first arrow in the diagram a big higher than the other two? 🤔 I would also maybe change:

Define an alert > `Define an alerting rule`

Author: I changed it into `Define an alerting rule` but the arrows are the same for sure - that weird 🤔

Reviewer: Maybe it's my 👁
